dynamodb/src/main/kotlin/com/playkids/onboarding/dynamodb/DynamoDBProfileDAO.kt

```diff
@@ -0,0 +1,118 @@
+package com.playkids.onboarding.dynamodb
+
+import com.movile.kotlin.commons.dynamodb.*
+import com.playkids.onboarding.core.model.Item
+import com.playkids.onboarding.core.model.ItemId
+import com.playkids.onboarding.core.model.Profile
+import com.playkids.onboarding.core.model.ProfileId
+import com.playkids.onboarding.core.persistence.ProfileDAO
+import com.playkids.onboarding.core.util.ChooseValue
+import com.playkids.onboarding.core.util.ItemsCurrency
+import com.playkids.onboarding.dynamodb.extensions.itemOrNull
+import com.playkids.onboarding.dynamodb.extensions.listOfString
+import com.playkids.onboarding.dynamodb.extensions.toListAttributeValue
+import com.typesafe.config.Config
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue
+import software.amazon.awssdk.services.dynamodb.model.UpdateItemRequest
+
+class DynamoDBProfileDAO(config: Config, private val dynamoDbClient: DynamoDbAsyncClient): ProfileDAO {
+
+    private val tableName = "onboarding-pedro-profile"
+
+    override suspend fun create(profile: Profile) {
+        dynamoDbClient.putItem {
+            it.tableName(tableName)
+                .conditionExpression("attribute_not_exists($ID)")
+                .item(profile.toItem())
+        }.awaitRaiseException()
+    }
+
+    override suspend fun find(id: ProfileId): Profile? =
+        this.query(id)?.toProfile()
+
+    override suspend fun getItemsAndCurrency(id: ProfileId, projection: Map<String, String>, currency: String): ItemsCurrency? =
+        this.query(id, projection)?.toItemsCurrency(currency)
+
+
+
+    override suspend fun addItem(profileId: ProfileId, item: List<ItemId>) {
+        dynamoDbClient.updateItem(
+            UpdateItemRequest.builder()
+                .tableName(tableName)
+                .key(mapOf(ID to profileId.toAttributeValue()))
+                .updateExpression("SET #items = list_append( #items, :val)")
+                .expressionAttributeNames(mapOf("#items" to ITEMS))
+                .expressionAttributeValues(mapOf(":val" to item.toListAttributeValue()))
+                .build()
+        )
+            .awaitRaiseException()
+    }
+
+    override suspend fun updateCurrency(profileId: ProfileId, operation: String, currency: String, chooseValue: ChooseValue) {
```

Comment:
In the Operation/Currency parameters, you could have used an enum. In the case of the operation, you could actually just ignore it and use the ADD method of dynamoDB, when you need to subtract, just add a negative amount :)